app/src/main/java/chat/rocket/android/helper/RocketChatUserAvatar.kt

```diff
@@ -1,17 +1,91 @@
 package chat.rocket.android.helper
 
+import android.content.Context
+import android.graphics.Typeface
+import android.graphics.drawable.Drawable
+import com.amulyakhare.textdrawable.TextDrawable
 import java.net.URLEncoder
 
-class RocketChatUserAvatar(val hostname: String, val username: String) {
-
-    val imageUri: String
-        /** REMARK
-         * This is often a SVG image (see Rocket.Chat:server/startup/avatar.js).
-         */
-        get() {
-            return "https://" +
-                    hostname.replace("http://", "").replace("https://", "") +
-                    "/avatar/" +
-                    URLEncoder.encode(username, "UTF-8")
+object RocketChatUserAvatar {
+
+    /**
+     * Returns the user avatar URI.
+     *
+     * REMARK: This is often a SVG image (Rocket.Chat:server/startup/avatar.js).
+     *
+     * @param hostname The server's hostname.
+     * @param username The username.
+     * @return The user avatar URI.
+     * @see getImageFormat
+     */
+    fun getUri(hostname : String, username: String): String {
+        return "https://" +
+                hostname.replace("http://", "").replace("https://", "") +
+                "/avatar/" +
+                URLEncoder.encode(username, "UTF-8")
+    }
+
+    /**
+     * Returns the user avatar image format.
+     *
+     * @param uri The user avatar image URI to get the image format.
+     * @return The user avatar image format. Possible values are: "image/jpeg", "image/png" and "image/svg+xml".
+     * @see getUri
+     */
+    fun getImageFormat(uri: String): String {
+        return OkHttpHelper.getContentType(uri)
+    }
+
+    /**
+     * Returns a drawable with username initials.
+     *
+     * @param username The username.
+     * @param context The context.
+     * @return A drawable with username initials.
+     * @see getUsernameInitials
+     */
+    fun getTextDrawable(username: String, context: Context): Drawable {
+        val round = (4 * context.resources.displayMetrics.density).toInt()
+
+        return TextDrawable.builder()
+                .beginConfig()
+                .useFont(Typeface.SANS_SERIF)
```

Comment:
It will be rendered as expected on Application, since the `Typeface` was [added](https://developer.android.com/reference/android/graphics/Typeface.html) by the Android platform since API level 1.